### PR TITLE
Yet another Migrate bevy_sprite to required components

### PR DESCRIPTION
**Pull request by KirmesBude**
# Objective

Migrate bevy_sprite Sprite to required components.
Alternative to https://github.com/bevyengine/bevy/pull/15489
It does not use any of the proposals from [here](https://hackmd.io/@bevy/required_components/) so feel free to disregard this PR.

## Solution

- Sprite renamed to SpriteProperties
- Introduce new wrapper component Sprite (contains Handle<Image>), that requires SpriteProperties, Transform, Visibility and SyncToRenderWorld
- Sprite specific usage of Handle<Image> replaced with Sprite 

Previously:
```rust
commands.spawn(SpriteBundle {
    texture: textures.add(Image::default()),
    sprite: Sprite {
        color: Color::srgb(7.5, 0.0, 7.5),
       ..default()
    },
    transform: Transform::from_translation(Vec3::new(-200., 0., 0.)),
    ..default()
});
```

Now:
```rust
commands.spawn((
    Sprite(textures.add(Image::default())),
    SpriteProperties {
        color: Color::srgb(7.5, 0.0, 7.5),
       ..default()
    },
    Transform::from_translation(Vec3::new(-200., 0., 0.)),
});
```

Maybe SpriteProperties should just be part of Sprite, not sure. I am open to changing that. Having a dedicated wrapper component for an asset is nice though, e.g. for change detection.
I did not touch TextureAtlas, but I can see the motivation for that in the context of editor usability/suggestion - I just do not think required components are the right tool for that.

## Testing
I ran all tests that I modified and everything seemed to work, but I did not compare to main.
bevy_sprite tests were also run and passed.

---

## Migration Guide

Asset handles for sprite images must now be wrapped in the Sprite component. Raw handles as components no longer render sprites. The component previously known as Sprite is now called SpriteProperties.

As a result SpriteBundle has been deprecated. Instead, use the Sprite components directly.

```rust
commands.spawn(SpriteBundle {
    texture: textures.add(Image::default()),
    sprite: Sprite {
        color: Color::srgb(7.5, 0.0, 7.5),
       ..default()
    },
    transform: Transform::from_translation(Vec3::new(-200., 0., 0.)),
    ..default()
});
```

Now:
```rust
commands.spawn((
    Sprite(textures.add(Image::default())),
    SpriteProperties {
        color: Color::srgb(7.5, 0.0, 7.5),
       ..default()
    },
    Transform::from_translation(Vec3::new(-200., 0., 0.)),
});
```

If you have any queries for sprites using Handle<Image> make sure to change them to Sprite.

### CODE REVIEW

File: examples/asset/asset_decompression.rs [hunk 1]
```diff
@@ -111,7 +111,7 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
             compressed: asset_server.load("data/compressed_image.png.gz"),
```
Comment (KirmesBude) [inline examples/asset/asset_decompression.rs:111]: Not sure how to handle this generically now.

Comment (KirmesBude) [reply examples/asset/asset_decompression.rs:115]: I added Handle<A>: Into<C> trait bounds. That means C needs to implement From<Handle<A>, which is true for Sprite (and Mesh as well).
May not be the best solution.

**Review by alice-i-cecile — APPROVED** [review]
I have a mild preference for this design. It's more flexible and the migration is easier. Either one is fine by me though.

**Review by wilk10 — APPROVED** [review]
Thanks for working on this. 

I very much prefer this implementation to the alternative PR. 

I know that there's no promise of stability but at least reducing churn for users when upgrading is an important factor to consider in this case, in my opinion. 

I particularly like that it keeps `Sprite` and `TextureAtlas` separate, like it was discussed at length, approved and merged with https://github.com/bevyengine/bevy/pull/5103 , whereas the blessed PR kinda reverts that change.

I'll try to argue my case on Discord, I guess.

Edit: [Discussion on Discord](https://discord.com/channels/691052431525675048/1264881140007702558/1293125823926374493)

**Comment by cart** [comment]
Responded with thoughts here: https://github.com/bevyengine/bevy/pull/15489#discussion_r1792604747
(please discuss there)

**Comment by alice-i-cecile** [comment]
Merging https://github.com/bevyengine/bevy/pull/15489 instead, per the discussion linked just above :)

**Pull request closed without merging**